homeassistant/components/media_player/__init__.py

```diff
@@ -753,24 +753,28 @@ def volume_up(self):
         if self.volume_level < 1:
             self.set_volume_level(min(1, self.volume_level + .1))
 
+    @asyncio.coroutine
     def async_volume_up(self):
-        """Turn volume up for media player.
-
-        This method must be run in the event loop and returns a coroutine.
-        """
-        return self.hass.loop.run_in_executor(None, self.volume_up)
+        """Turn volume up for media player."""
+        try:
+            yield from self.hass.loop.run_in_executor(None, self.volume_up)
+        except NotImplementedError:
+            yield from self.async_set_volume_level(
```

Comment:
Because set_volume_level and volume_up is't the same thing. ex. In Denon my reciver, the volume range is from -80 to +10, so 90 "real steps". The native volume stepping commands step one "real" step , while set_volume_level steps 10%, aka 9 steps. 9 "real steps" is quite a big volume difference.

If we're going to make async_volume_{up,down} call {async_,}set_volume_level we should drop volume_{up,down} to, because those wouldn't make sense then.